packages/repack/src/modules/ScriptManager/ScriptManager.ts

```diff
@@ -272,6 +272,28 @@ export class ScriptManager extends EventEmitter {
       const script = Script.from({ scriptId, caller }, locator, false);
       const cacheKey = `${scriptId}_${caller ?? 'unknown'}`;
 
+      // Check if user provided a custom shouldUpdateScript function
+      if (locator.shouldUpdateScript) {
+        // If so, we need to wait for it to resolve
+        const fetch = await locator.shouldUpdateScript(
```

Comment:
Despite the same name of parameter under`srcipt.locator` object, I would suggest changing the name to something more meaningful. E.g. I thought at first glance that it's a method not a boolean.

Proposed name is eg `shoudlFetchScript`

@RafikiTiki wdyt ?